src/Diagnostics/horizontal_average.jl

```diff
@@ -3,12 +3,42 @@ using Oceananigans.BoundaryConditions
 using Oceananigans.Utils
 using Oceananigans.Grids: total_size
 
+"""
+    VolumeAverage{F, R, P, I, Ω} <: AbstractDiagnostic
+
+A diagnostic for computing the volume average of a field.
+"""
+mutable struct VolumeAverage{F, R, P, I, Ω, G} <: AbstractAverage
```

Comment:
The name of this file needs to be changed to `averages.jl` because there is more than just `HorizontalAverage` in it.